addons/web/static/src/views/fields/many2many_tags/many2many_tags_field.js

```diff
@@ -143,9 +146,14 @@ export class Many2ManyTagsField extends Component {
     }
 
     get tags() {
-        return this.props.record.data[this.props.name].records.map((record) =>
-            this.getTagProps(record)
-        );
+        // Use Set to remove client-side duplicates of selected tags, ensuring each tag is displayed only once.
+        return [
+            ...new Set(
+                this.props.record.data[this.props.name].records.map((record) =>
```

Comment:
there should be no duplicate in this.props.record.data[this.props.name].records (the model is the source of truth, it should be correct)